packages/std-client/src/setup/setupMUDV2Network.ts

```diff
@@ -107,11 +107,12 @@ export async function setupMUDV2Network<C extends ContractComponents, S extends
 
   // Function to register new components in mappings object
   function registerComponent(key: string, component: ContractComponent) {
-    const { contractId, tableId } = component.metadata;
-    if (tableId) {
-      mappings[tableId] = key;
+    if (typeof component.metadata?.tableId === "string") {
+      mappings[component.metadata.tableId] = key;
     } else {
-      mappings[keccak256(contractId)] = key;
+      mappings[
+        keccak256(typeof component.metadata?.contractId === "string" ? component.metadata.contractId : component.id)
```

Comment:
would this be equivalent?
```suggestion
        keccak256(component.metadata?.contractId ?? component.id)
```

Reply:
I had this before, but TS complained about the `contractId` value being `{}` or something? Might be related to `Metadata` default being `Record<string, unknown>`